src/main/java/data/AnalysisResults.java

```diff
@@ -36,4 +39,41 @@ public String getRawText() {
   public ImmutableSet<String> getCategories() {
     return categories;
   }
-}
\ No newline at end of file
+
+  public Optional<String> getStore() {
+    return store;
+  }
+
+  public static class Builder {
+    private final String rawText;
+    private ImmutableSet<String> categories;
+    private Optional<String> store;
+
+    public Builder(String rawText) {
+      this.rawText = rawText;
+    }
+
+    public String getRawText() {
+      return rawText;
+    }
+
+    public Builder setCategories(Set<String> categories) {
+      this.categories = ImmutableSet.copyOf(categories);
+      return this;
+    }
+
+    public Builder setStore(String store) {
+      this.store = Optional.of(store);
+      return this;
+    }
+
+    public AnalysisResults build() {
+      // No logo was detected.
+      if (store == null) {
+        store = Optional.empty();
+      }
```

Comment:
nit: you can use `Optional.ofNullable()` - Returns an Optional describing the specified value, if non-null, otherwise returns an empty Optional.